### PR TITLE
Generate 2bpp fonts, with equivalent glyph decode stage at runtime.

### PR DESCRIPTION
**Pull request by tzarc**
As per #95, as a first cut, just for fonts. Saves about 15.6k of flash. No noticeable speed decrease on my setup.

before:

    % arm-none-eabi-readelf  -l opendps/opendps.elf

    Elf file type is EXEC (Executable file)
    Entry point 0x8006e7d
    There are 6 program headers, starting at offset 52

    Program Headers:
      Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
      LOAD           0x000000 0x08000000 0x08000000 0x0de28 0x0de28 R E 0x10000
      LOAD           0x010150 0x20000150 0x0800de28 0x0083c 0x0083c RW  0x10000
      LOAD           0x010990 0x20000990 0x0800e664 0x00000 0x007d4 RW  0x10000
      LOAD           0x01f800 0x0800f800 0x0800f800 0x00000 0x00800 RW  0x10000
      LOAD           0x020000 0x20000000 0x20000000 0x00000 0x00150 RW  0x10000
      LOAD           0x011ff0 0x20001ff0 0x20001ff0 0x00000 0x00010 RW  0x10000

     Section to Segment mapping:
      Segment Sections...
       00     .text
       01     .data
       02     .bss
       03     .past
       04     .ram_vect
       05     .bootcom

after:

    % arm-none-eabi-readelf  -l opendps/opendps.elf

    Elf file type is EXEC (Executable file)
    Entry point 0x8006e95
    There are 6 program headers, starting at offset 52

    Program Headers:
      Type           Offset   VirtAddr   PhysAddr   FileSiz MemSiz  Flg Align
      LOAD           0x000000 0x08000000 0x08000000 0x0a074 0x0a074 R E 0x10000
      LOAD           0x010150 0x20000150 0x0800a074 0x007a4 0x007a4 RW  0x10000
      LOAD           0x0108f8 0x200008f8 0x0800a818 0x00000 0x007d4 RW  0x10000
      LOAD           0x01f800 0x0800f800 0x0800f800 0x00000 0x00800 RW  0x10000
      LOAD           0x020000 0x20000000 0x20000000 0x00000 0x00150 RW  0x10000
      LOAD           0x011ff0 0x20001ff0 0x20001ff0 0x00000 0x00010 RW  0x10000

     Section to Segment mapping:
      Segment Sections...
       00     .text
       01     .data
       02     .bss
       03     .past
       04     .ram_vect
       05     .bootcom

size reduction of .text+.data sections:

    0xE664 -> 0xA818 bytes
    58,980 -> 43,032 bytes
    15,948 bytes reduction

### CODE REVIEW

**Comment by Xenoamor** [comment]
This looks seriously good. That's some huge savings in FLASH space. I'll have a test of this when I get back from work

**Comment by Xenoamor** [comment]
Out of interest how does the lookup table work? I guess it's converting 2bpp to the bgr565 used on the screen but I don't quite understand how

I'm really liking how you're now handling the character to glyph assignment now using 0 size characters by the way

File: opendps/gfx_lookup.c [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include "gfx_lookup.h"
+
+const uint32_t mono2bpp_lookup[16] = {
```
Comment (kanflo) [inline opendps/gfx_lookup.c]: This one could perhaps be explained.

**Comment by kanflo** [comment]
Great work @tzarc !

**Comment by tzarc** [comment]
I'll make a supplemental PR with comments on the lookup table. Will be more likely to be seen if it's in the code than some random PR comments for something that was closed.

**Pull request merged**